src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,152 @@
+            resultSet.next();
+            cart.setId(resultSet.getLong(1));
+            insertCartsProducts(cart);
+            return cart;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create cart", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM carts WHERE cart_id = ?;";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                ShoppingCart shoppingCart = getShoppingCart(resultSet);
+                shoppingCart.setProducts(getProductsFromShoppingCartId(id));
+                return Optional.of(shoppingCart);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to get cart", e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM carts;";
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            List<ShoppingCart> shoppingCarts = new ArrayList<>();
+            while (resultSet.next()) {
+                ShoppingCart shoppingCart = getShoppingCart(resultSet);
+                shoppingCart.setProducts(getProductsFromShoppingCartId(shoppingCart.getId()));
+                shoppingCarts.add(shoppingCart);
+            }
+            return shoppingCarts;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to get all carts", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart cart) {
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "UPDATE carts SET user_id = ? "
+                    + "WHERE cart_id = ?;";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, cart.getUserId());
+            statement.setLong(2, cart.getId());
+            statement.executeUpdate();
+            deleteShoppingCartFromCartsProducts(cart.getId());
+            insertCartsProducts(cart);
+            return cart;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to update", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteShoppingCartQuery = "DELETE FROM carts WHERE cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            deleteShoppingCartFromCartsProducts(id);
+            PreparedStatement statement = connection.prepareStatement(deleteShoppingCartQuery);
```

Comment:
```suggestion
            var statement = connection.prepareStatement(deleteShoppingCartQuery);
```
can we use var here? line will be a bit shorter